vmdb/app/models/metric/aggregation.rb

```diff
@@ -1,98 +1,109 @@
 module Metric::Aggregation
-  def self.aggregate_for_column(*args)
-    self.execute_for_column(:aggregate, *args)
-  end
+  module Aggregate
+    def self.average(col, obj, result, counts, value)
+      return if value.nil?
+      result[col] += value
+      counts[col] += 1
```

Comment:
It doesn't directly.  An average is a summation followed by a division.  This method (the aggregator) does the summation.  The division is done in the processor.